docs/docs/neptune_for_amundsen.md

```diff
@@ -0,0 +1,152 @@
+
+
+### Create user(s) and give it(them) access to Neptune and S3
+
+The metadata and dataloader processes require access to Neptune and S3.  In the most direct setup,
+each would use a distinct IAM user and receive just those permissions required to access Neptune
+and S3.  An another, you might run the metadata service and dataloader processes in EC2 and you
+would grant whatever Role they assume the required permissions for Neptune and S3.
+
+The terraform example creates a Policy which gives the metadata and dataloader services sufficient
+access, and creates 2 Users and gives attaches the Policy.
+
+
+### Create public endpoint for Neptune
+
+Neptune does not have public endpoints.  The state of the art seems to be using
+a [loadbalancer](
+https://github.com/aws-samples/aws-dbs-refarch-graph/tree/master/src/connecting-using-a-load-balancer)
+as well as glue to allow the mismatch in hostnames.
+
+
+The terraform sets up a publicly available NLB -- albeit statically configured, so is only suitable
+for development or other deployments where one could tolerate downtime as instances are
+reconfigured.  (Or you could apply the CloudWatch + Lambda approach and minimize that downtime.)
+
+
+### Configure amundsen-gremlin
+
+amundsen-gremlin needs to know about some of the resources you configured:
+
+* `AWS_REGION_NAME`: name of the AWS region in which you created the Neptune instance
+* `NEPTUNE_BULK_LOADER_S3_BUCKET_NAME`: name of the bucket to use for the Neptune Bulk Loader input
+* `NEPTUNE_URL`: either 
+  * the URL for the /gremlin endpoint of the Neptune cluster (or instance)
+    e.g. `something.cluster-xxxx.region.amazonaws.com:8182/gremlin`, 
+  * or if you use the NLB approach, a dict like 
+  ```
+  {
+    "neptune_endpoint": "something.cluster-xxxx.region.amazonaws.com",
+    "neptune_port": 8182,
+    "uri": "wss://otherthing-yyyyy.elb.region.amazonaws.com:8182/gremlin"
+  }  
+  ```
+* `NEPTUNE_SESSION`: this depends on how you are authenticating your procesess, it could be
+  something like ```
+  boto3.session.Session(profile_name='youruserprofilehere',
+  region_name=AWS_REGION_NAME)```.  
+  Or suppose you have some secrets management service and read the file as `secrets_file_yaml` 
+  (certainly don't commit that file with your source), it could be something like:
+  ```
+  boto3.session.Session(
+      region_name=secrets_file_yaml['service_region'],
+      aws_access_key_id=secrets_file_yaml['aws_access_key_id'],
+      aws_secret_access_key=secrets_file_yaml['aws_secret_access_key'])
+  ```
+
+
+## Other Neptunic Considerations
+
+
+## IAM authentication
+
+You should use IAM authentication.  However, it comes with some mild complications.  The gremlin
+transport is usually websockets, and the requests-aws4auth library we use elsewhere is for requests,
+which does not support websockets at all.  In the [amazon-neptune-tools](
```

Comment:
I think either "the" from "In the amazon-neptune-tools, there is..." sentence should be dropped, or the word "repo" added at the end (e.g. "in the amazon-neptune-tools repo, there is..."

```suggestion
which does not support websockets at all.  In [amazon-neptune-tools](
```